## Project: CEERS AGN Search

*Name:* Thresa Kelly

*Professor:* Dr. Allison Kirkpatrick

*Date:* 12/2/21

*Science Goal:* Which galaxies in the EGS field are likely AGN, and what are their physical properties?

*Details:* CEERS is an approved project on JWST to observe the Extended Groth Strip with the NIRCam, NIRSpec, and MIRI instruments. Thresa is going to identify AGN in the MIRI field of view. She will plot their spectral energy distributions, look for spectra in the 3D-HST database, look for radio observations through the FIRST database, and make images.



In [9]:
# imports 
from astropy.io import ascii
import numpy as np 

## Milestone: Read the .csv file

In [16]:
# read data from file 
data = ascii.read("EGS_for_Thresa.csv")

# view data
print(data.columns)
print(data)

<TableColumns names=('ID','RA','Dec','Redshift','u','u Unc','g','g Unc','r','r Unc','i','i Unc','z','z Unc','y','y Unc','IRAC Ch 1','Ch 1 Unc','IRAC Ch 2','Ch 2 Unc','IRAC Ch 3','Ch 3 Unc','IRAC Ch 4','Ch 4 Unc','PACS 100','100 Unc','PACS 160','160 Unc','SPIRE 250','250 Unc','SPIRE 350','350 Unc','SPIRE 500','500 Unc')>
             ID                 RA        Dec    ... SPIRE 500  500 Unc 
--------------------------- ---------- --------- ... --------- ---------
HELP_J141411.189+520025.636 213.546622 52.007121 ...       NaN       NaN
HELP_J141411.417+515923.419 213.547572 51.989839 ...       NaN       NaN
HELP_J141411.539+515959.135 213.548077  51.99976 ...       NaN       NaN
HELP_J141411.902+515927.909 213.549592 51.991086 ...       NaN       NaN
HELP_J141412.159+515941.025 213.550661 51.994729 ...       NaN       NaN
HELP_J141412.263+515904.194 213.551097 51.984498 ...       NaN       NaN
HELP_J141412.597+520041.294 213.552488 52.011471 ...  7254.822   5038.43
HELP_J141412.781+5159

## Milestone: Which wavelengths correspond with which filters in the .csv file?



'u' = ultraviolet: λeff= 365 nm, Δλ= 66 nm

'g' = green: λeff= 464 nm, Δλ= 128 nm

'r' = red: λeff= 658 nm, Δλ= 138 nm

'i' = infared: λeff= 806 nm, Δλ= 149 nm

'z' = infared: λeff= 900 nm

'y' = infared: λeff= 1020 nm, Δλ= 120 nm

'IRAC Ch 1' =  Infrared Array Camera channel 1: λcenter= 3.56 um

'IRAC Ch 2' =  Infrared Array Camera channel 2: λcenter= 4.54 um

'IRAC Ch 3' =  Infrared Array Camera channel 3: λcenter= 5.73 um

'IRAC Ch 4' =  Infrared Array Camera channel 4: λcenter= 7.91 um

'PACS 100' = Photodetector Array Camera and Spectrometer: 100 um

'PACS 160' = Photodetector Array Camera and Spectrometer: 160 um

'SPIRE 250' = Photometer: 250 um

'SPIRE 350' = Photometer: 350 um

'SPIRE 500' = Photometer: 500 um

* Filters: https://en.wikipedia.org/wiki/Photometric_system
* IRAC paper: https://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.145.2828&rep=rep1&type=pdf#:~:text=IRAC%20is%20a%20four%2Dchannel,%3B%204.5%20and%208%20m).
* PACS: https://www.cosmos.esa.int/web/herschel/pacs-overview 
* Spire: https://www.cosmos.esa.int/web/herschel/first-light

## Milestone: Select galaxies in the MIRI field of view

Tips:
* use tools from class project
* CEERS proposals → look at tables in front
* table lists RA and DEC of 4 MIRI pointings (ex EGS7)
* MIRI FOV = 113 as (?)
* calculate angular distance of object to center of MIRI pointing. if <1.9am, keep object. if >1.9am, throw out object.
* .csv RA and DEC already in decimal degrees


In [14]:
# the four MIRI pointings:
#   MIRI-EGS7   RA= 14 20 14.6913  DEC= +52 55 25.43
#   MIRI-EGS8   RA= 14 19 57.9690  DEC= +52 55 51.53
#   MIRI-EGS9   RA= 14 19 52.2788  DEC= +52 51 40.09
#   MIRI-EGS10  RA= 14 19 35.6531  DEC= +52 52 6.56

miri_7_ra  = (14.*15.) + (20./4.) + (14.6913/240.)   # hrs*15 + minutes/4 + seconds/240
miri_7_dec = +( 52. + (55./60) + (25.43/3600) )      # + or -  ( degrees + minutes/60 + seconds/3600 )

miri_8_ra  = (14.*15.) + (19./4.) + (57.9690/240.)
miri_8_dec = +( 52. + (55./60) + (51.53/3600) )

miri_9_ra  = (14.*15.) + (19./4.) + (52.2788/240.)
miri_9_dec = +( 52. + (51./60) + (40.09/3600) )

miri_10_ra  = (14.*15.) + (19./4.) + (35.6531/240.)
miri_10_dec = +( 52. + (52./60) + (6.56/3600) )

# Add all miri ra and dec into arrays
miri_ra  = [miri_7_ra, miri_8_ra, miri_9_ra, miri_10_ra]
miri_dec = [miri_7_dec, miri_8_dec, miri_9_dec, miri_10_dec]

print(miri_ra)
print(miri_dec)

# find average position
miri_ra_avg = np.average(miri_ra)
miri_dec_avg = np.average(miri_dec)

print(miri_ra_avg)
print(miri_dec_avg)

# MIRI field of view is 113 as = 0.0313889 deg
miri_fov = 0.0313889 


[215.06121375, 214.9915375, 214.96782833333333, 214.89855458333332]
[52.92373055555555, 52.93098055555555, 52.861136111111115, 52.86848888888889]
214.97978354166665
52.896084027777775


In [12]:
# Function below by Dr. Kirkpatrick from Age-Of-Stellar-Clusters project

# define a function that takes the RA and DEC locations of two objects, and returns the distance between them
def distance_along_sky(raobject1, decobject1, raobject2, decobject2):
    # inputs in degrees
    # using the distance along a great circle equation from http://spiff.rit.edu/classes/phys301/lectures/precession/precession.html#sep
    # Note, cosine and sine in python use RADIANS as default units
    cosy = np.cos(  np.deg2rad(90.-decobject1) )* np.cos(  np.deg2rad(90.-decobject2)  )  +     np.sin(  np.deg2rad(90.-decobject1)  )*np.sin(  np.deg2rad(90.-decobject2))*np.cos(  np.deg2rad(raobject1-raobject2))
    angular_distance_radians= np.arccos(cosy)
    
    #outputs in radians, so let's convert back to degrees to make it easier for us later
    angular_distance_degrees=np.rad2deg(angular_distance_radians)
    
    #this function will print out one value, the distance between two points on a great circle in degrees
    return angular_distance_degrees